assets/js/components/data.js

```diff
@@ -356,79 +369,94 @@ const data = {
 			return;
 		}
 
-		if ( window.sessionStorage ) {
-			const toStore = {
-				value,
-				date: Date.now() / 1000
-			};
-			setCache( 'sessionStorage', type + '::' + key, JSON.stringify( toStore ) );
+		lazilySetupLocalCache();
+
+		googlesitekit.admin.datacache[ key ] = data;
+
+		if ( ! window.sessionStorage ) {
+			return;
 		}
+
+		const toStore = {
+			value: data,
+			date: Date.now() / 1000
+		};
+		window.sessionStorage.setItem( 'googlesitekit_' + key, JSON.stringify( toStore ) );
 	},
 
 	/**
 	 * Gets data from the cache.
 	 *
-	 * @param {object} type   The data object.
 	 * @param {string} key    The cache key,
 	 * @param {int}    maxAge The cache TTL in seconds.
 	 *
 	 * @return {mixed} Cached data, or false if lookup failed.
 	 */
-	getCache( type, key, maxAge, hashlessKey = '' ) {
+	getCache( key, maxAge ) {
 
 		// Skip if js caching is disabled.
 		if ( googlesitekit.admin.nojscache ) {
 			return false;
 		}
 
-		// Require sessionStorage for local caching.
+		lazilySetupLocalCache();
+
+		// Check variable cache first.
+		if ( 'undefined' !== typeof googlesitekit.admin.datacache[ key ] ) {
+			return googlesitekit.admin.datacache[ key ];
+		}
+
 		if ( ! window.sessionStorage ) {
 			return false;
 		}
 
-		const cache = JSON.parse( window.sessionStorage.getItem( type + '::' + key ) );
-		const datacache = googlesitekit.admin.datacache && JSON.parse( googlesitekit.admin.datacache );
+		// Check persistent cache.
+		const cache = JSON.parse( window.sessionStorage.getItem( 'googlesitekit_' + key ) );
+		if ( cache && 'object' === typeof cache && cache.date ) {
 
-		// Check the datacache, which only includes fresh data at load.
-		if ( ! cache && datacache && datacache[ 'googlesitekit_' + key ] ) {
-			return datacache[ 'googlesitekit_' + key ];
-		}
+			// Only return value if no maximum age given or value is newer than it.
+			if ( ! maxAge || ( Date.now() / 1000 ) - cache.date < maxAge ) {
 
-		// Check the datacache, which only includes fresh data at load.
-		if ( ! cache && datacache && datacache[ 'googlesitekit_' + hashlessKey ] ) {
-			return datacache[ 'googlesitekit_' + hashlessKey ];
-		}
+				// Set variable cache.
+				googlesitekit.admin.datacache[ key ] = cache.value;
 
-		if ( ! cache ) {
-			return false;
+				return cache.value;
+			}
 		}
 
-		const age = ( Date.now() / 1000 ) - cache.date;
-		if ( age < maxAge ) {
-			return cache.value;
-		}
 		return false;
 	},
 
 	/**
 	 * Removes data from the cache.
 	 *
-	 * @param {object} type The data object.
-	 * @param {string} key  The cache key,
+	 * @param {string} key The cache key,
 	 */
-	deleteCache( type, key ) {
-		if ( window.sessionStorage ) {
-			window.sessionStorage.removeItem( type + '::' + key );
+	deleteCache( key ) {
+		lazilySetupLocalCache();
+
+		if ( 'undefined' !== typeof googlesitekit.admin.datacache[ key ] ) {
+			delete googlesitekit.admin.datacache[ key ];
+		}
```

Comment:
This conditional is unnecessary. It's safe to delete an undefined key as long as `datacache` is established.